spark/src/test/java/io/spark/ddf/content/ViewHandlerTest.java

```diff
@@ -31,4 +34,53 @@ public void testRemoveColumns() throws DDFException {
     Assert.assertEquals(27, newddf2.getNumColumns());
     Assert.assertEquals(26, newddf3.getNumColumns());
   }
+
+  @Test
+  public void testSubsettingWithGrep() throws DDFException {
+    createTableAirline();
+
+    DDF ddf = manager.sql2ddf("select * from airline");
+
+    List<Column> columns = Lists.newArrayList();
+    Column col = new Column();
+    col.setName("origin");
+    columns.add(col);
+
+    Operator grep = new Operator();
+    grep.setName(OperationName.grep);
+    Expression[] operands = new Expression[2];
+    StringVal val = new StringVal();
+    val.setValue("IAD");
+    operands[0] = val;
+    operands[1] = col;
+    grep.setOperarands(operands);
+
+    DDF ddf2 = ddf.VIEWS.subset(columns, grep);
+    Assert.assertEquals(2, ddf2.getNumRows());
+
+  }
+
+  @Test
+  public void testSubsettingWithGrepIgnoreCase() throws DDFException {
+    createTableAirline();
+
+    DDF ddf = manager.sql2ddf("select * from airline");
+
+    List<Column> columns = Lists.newArrayList();
+    Column col = new Column();
+    col.setName("origin");
+    columns.add(col);
+
+    Operator grep = new Operator();
+    grep.setName(OperationName.grep_ic);
+    Expression[] operands = new Expression[2];
+    StringVal val = new StringVal();
+    val.setValue("iad");
+    operands[0] = val;
+    operands[1] = col;
+    grep.setOperarands(operands);
+
+    DDF ddf2 = ddf.VIEWS.subset(columns, grep);
```

Comment:
This is in PyClient https://github.com/adatao/PyClient/blob/7d1d9d999c5dcc0efb6a67d919df7019083cc5b0/pAnalytics/adatao/core/subset_expression.py#L306